web/src/stream_events.ts

```diff
@@ -28,20 +28,47 @@ import * as stream_muting from "./stream_muting.ts";
 import * as stream_settings_api from "./stream_settings_api.ts";
 import * as stream_settings_ui from "./stream_settings_ui.ts";
 import * as sub_store from "./sub_store.ts";
+import type {StreamSubscription} from "./sub_store.ts";
 import * as unread_ui from "./unread_ui.ts";
 import * as user_profile from "./user_profile.ts";
 
 // In theory, this function should apply the account-level defaults,
 // however, they are only called after a manual override, so
 // doing so is unnecessary with the current code.  Ideally, we'd do a
 // refactor to address that, however.
-function update_stream_setting(sub, value, setting) {
-    const $setting_checkbox = $(`#${CSS.escape(setting)}_${CSS.escape(sub.stream_id)}`);
+function update_stream_setting(
+    sub: StreamSubscription,
+    value: boolean | null,
+    setting:
+        | "desktop_notifications"
+        | "audible_notifications"
+        | "push_notifications"
+        | "email_notifications"
+        | "wildcard_mentions_notify"
+        | "pin_to_top",
+): void {
+    const $setting_checkbox = $(`#${CSS.escape(setting)}_${CSS.escape(sub.stream_id.toString())}`);
     $setting_checkbox.prop("checked", value);
+    if (setting === "pin_to_top") {
+        assert(value !== null);
+        sub[setting] = value;
+        return;
+    }
     sub[setting] = value;
 }
 
-export function update_property(stream_id, property, value, other_values) {
+type UpdatableStreamProperties = sub_store.ApiStreamSubscription & {in_home_view: boolean};
```

Comment:
Probably there's a follow-up to remove `in_home_view`; we changed the API to use its more rationally named negation `is_muted` a year ago or something, so there's no reason the web app should need to mention the legacy API name.
```
$ git grep in_home_view web/src/
web/src/stream_data.ts:export function all_subscribed_streams_are_in_home_view(): boolean {
web/src/stream_events.js:        case "in_home_view":
web/src/stream_events.js:            // Legacy in_home_view events are only sent as duplicates of
web/src/stream_list.ts:export function set_in_home_view(stream_id: number, in_home: boolean): void {
web/src/stream_muting.ts:    stream_list.set_in_home_view(sub.stream_id, !sub.is_muted);
```

Reply:
Yeah I didn't feel like I had enough context here to be confident about removing it.